opendrift/models/bivalvelarvae.py

```diff
@@ -326,7 +357,32 @@ def interact_with_coastline(self,final = False):
                                          (self.elements.age_seconds >= self.get_config('drift:min_settlement_age_seconds')),
                                          reason='settled_on_coast')
 
-
+                
+    def interact_with_habitat(self):
+           """Habitat interaction according to configuration setting
+               The method checks if a particle is within the limit of an habitat before to allow settlement
+           """        
+           # Get age of particle
+           old_enough = np.where(self.elements.age_seconds >= self.get_config('drift:min_settlement_age_seconds'))[0]
+           if len(old_enough) > 0 :
+               pts_lon = self.elements.lon[old_enough]
+               pts_lat = self.elements.lat[old_enough]
+               # Check if position of particle is within boundaries of polygons
+               for i in range(len(pts_lon)):
+                   pt = Point(pts_lon[i], pts_lat[i])
+                   for index, item in enumerate(shp):
+                       pts = bins[index].points # Access one polygon
+                       poly = Polygon(pts)
+                       in_habitat = pt.within(poly)
+                       if in_habitat == True:
+                           #import pdb; pdb.set_trace()
+                           self.environment.land_binary_mask[old_enough[i]] = 1
+                           
+           # Deactivate elements that are within a polygon and old enough to settle
+           # ** function expects an array of size consistent with self.elements.lon                
+           self.deactivate_elements((self.environment.land_binary_mask == 1), reason='settled_in_habitat')
```

Comment:
# This should work better:
self.deactivate_elements((self.environment.land_binary_mask == 6), reason='settled_in_habitat')